### ISSUE TITLE
Re-think concept of "Station for Current Weather State"

### ISSUE THREAD

**Issue by dbrgn** [izacus/hass-swissweather#5]
First of all, thanks for writing this integration, it worked great out-of-the-box!

When setting up the integration, one must choose the "station for current weather state". Unfortunately this causes some issues for me:

- The closest station for measuring the current temperature is "Lachen". However, that station does not provide any wind, rain or pressure masurements.
- The closest station for measuring the current wind speeed is "Schmerikon". However, that station does not provide any temperature or rain measurements.
- The closest station for measuring rain is "Jona". However, that station does not provide any temperature, wind or pressure measurements.

I could switch to the closest station providing all these measurements, which would be Hörnli (it's close, but on a hill, so measurements won't be representative for my location) or Wädenswil (more representative, but currently more than 1°C off from values at my place).

An idea how this could be solved:

- Pick separate stations for all measurement types when setting up the integration
- Filter the dropdown menu entries, so that only applicable stations can be selected
- Sort dropdown menu entrires by distance, so that the closest station is at the top (possibly selected by default)

Alternatively, completely decouple measurements from forecasts. When setting up the integration, allow adding forecast locations (by PLZ/ZIP) or measurement stations (from a dropdown). More effort by the user, but more flexible.

**Comment by izacus** [comment]
I agree, the way how measurement stations work right now is very clunky and not as practical as I'd like. Unfortunately, the refactor to use multiple stations is rather big so I'm not going to promise any movement in this direction soon (unless someone contributes :)

**Comment by dbrgn** [comment]
Thanks for your feedback 🙂 I _would_ like to contribute, but unfortunately I'm very tight on time (and am maintaining too many open source projects already), so I can't promise anything at the moment.
